pkg/process/workload.go

```diff
@@ -11,22 +11,26 @@ import (
 	"regexp"
 	"strings"
 
+	"github.com/cilium/tetragon/pkg/k8s/apis/cilium.io/v1alpha1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 var cronJobNameRegexp = regexp.MustCompile(`(.+)-\d{8,10}$`)
 
 // GetWorkloadMetaFromPod heuristically derives workload metadata from the pod spec.
-func getWorkloadMetaFromPod(pod *corev1.Pod) (metav1.ObjectMeta, metav1.TypeMeta) {
+func GetWorkloadMetaFromPod(pod *corev1.Pod) (v1alpha1.WorkloadObjectMeta, metav1.TypeMeta) {
 	if pod == nil {
-		return metav1.ObjectMeta{}, metav1.TypeMeta{}
+		return v1alpha1.WorkloadObjectMeta{}, metav1.TypeMeta{}
 	}
 	// try to capture more useful namespace/name info for deployments, etc.
 	// TODO(dougreid): expand to enable lookup of OWNERs recursively a la kubernetesenv
-	deployMeta := pod.ObjectMeta
-	deployMeta.ManagedFields = nil
-	deployMeta.OwnerReferences = nil
+	deployMeta := v1alpha1.WorkloadObjectMeta{
+		Name:        pod.GetObjectMeta().GetName(),
+		Namespace:   pod.GetObjectMeta().GetNamespace(),
+		Labels:      pod.GetObjectMeta().GetLabels(),
+		Annotations: pod.GetObjectMeta().GetAnnotations(),
```

Comment:
hmm, so Labels and Annotations are set to the pod metadata here, and kinda pretending to be the workload metadata if I understand correctly. It's actually no different from the current behaviour, but I'm wondering if these fields are needed at all?

Reply:
yeah let's get rid of them, i don't need them. we can always add them back later if we want it 🥰